.github/workflows/test.yml

```diff
@@ -0,0 +1,49 @@
+---
+name: CI on push release tag
+on:
+  pull_request:
+    branches:
+      - 'main'
+jobs:
+  run-tests:
+    name: Run DTS tests
+    if: ${{ github.head_ref == 'develop' }}
+    runs-on:
+      labels: dts-builder
+    steps:
+      - name: Checkout OSFV repo
+        shell: bash
+        run: |
+          git clone https://github.com/Dasharo/open-source-firmware-validation.git
+      - name: Install requirements
+        shell: bash
+        run: |
+          cd open-source-firmware-validation
+          git submodule update --init --checkout
+          python3 -m virtualenv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+      - name: Run QEMU
+        shell: bash
+        id: run_qemu
+        run: |
+          ./scripts/ci/qemu-run.sh nographic firmware &
+          pid=$!
+          echo "qemu_pid=$pid" >> "$GITHUB_OUTPUT"
+      # TODO: Add DPP keys to secrets and use them in tests
+      - name: Run tests
+        shell: bash
+        env:
+          DPP_EMAIL: ${{secrets.DPP_EMAIL}}
+          DPP_PASSWORD: ${{secrets.DPP_PASSWORD}}
+        run: |
+          robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no \
+             dts/dts-e2e.robot > /tmp/dts-test-ci
```

Comment:
Don't we want to see output?

Reply:
No, because in the future we will use DPP keys here and they may leak in case of errors.

Reply:
https://github.com/Dasharo/meta-dts/compare/df52aad37e5be1793811bae2603d8c3daec0688f..21a11dd5a691bfea6616bb097deeb37071100320

Reply:
1. Change settings so only 3mdeb team can see actions output
2. Maybe grep only titles?

    ```text
    robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no dasharo-compatibility/network-boot.robot | grep "| SUCCESS |\|| FAIL |"
    PXE001.001 Dasharo Network Boot is available :: This test aims to ... | FAIL |
    PXE002.001 Dasharo network boot menu boot options order is correct... | FAIL |
    PXE003.001 Autoboot option is available and works correctly :: Thi... | FAIL |
    PXE004.001 DTS option is available and works correctly :: This tes... | FAIL |
    PXE005.001 OS installation option is available and works correctly... | FAIL |
    PXE006.001 iPXE shell option is available and works correctly :: T... | FAIL |
    PXE007.001 Dasharo Network Boot over https not http :: This test a... | FAIL |
    Network-Boot                                                          | FAIL |
    ```

Reply:
This will save the full output to a file, while printing the parts matched by grep to stdout 84c53bfe49127ddbdc5603be301e64e434cd8eaf

Reply:
```diff
-dts/dts-e2e.robot | tee $LOG_DIR/output.log | grep "| SUCCESS |\|| FAIL |"
+dts/dts-e2e.robot | tee $LOG_DIR/output.log | grep "| PASS |\|| FAIL |"
```

Didn't check what passing test outputs only failing ones

Reply:
And you might still want to redirect stderr (I'm not sure if there might be sensitive data there)

Reply:
https://github.com/Dasharo/meta-dts/pull/193/commits/980b61335521194f4db526bdfe21eecb289116e3